pymeasure/instruments/rohdeschwarz/fsseries.py

```diff
@@ -0,0 +1,409 @@
+        # There is no space in the set commands, so we have to add it
+        return " " + str(value)
+
+
+class FSSeries(SCPIMixin, Instrument):
+    """
+    Represents a Rohde&Schwarz FS Series of spectrum analyzers like FSL and FSW.
+
+    All physical values that can be set can either be as a string of a value and a unit (e.g.
+    "1.2 GHz") or as a float value in the base units (Hz, dBm, etc.).
+    """
+
+    def __init__(self, adapter, name="Rohde&Schwarz FS series instrument", **kwargs):
+        super().__init__(adapter, name, **kwargs)
+
+    # Frequency settings ---------------------------------------------------------------------------
+
+    freq_span = Instrument.control(
+        "FREQ:SPAN?",
+        "FREQ:SPAN %s",
+        "Control frequency span in Hz.",
+    )
+
+    freq_center = Instrument.control(
+        "FREQ:CENT?",
+        "FREQ:CENT %s",
+        "Control center frequency in Hz.",
+    )
+
+    freq_start = Instrument.control(
+        "FREQ:STAR?",
+        "FREQ:STAR %s",
+        "Control start frequency in Hz.",
+    )
+
+    freq_stop = Instrument.control(
+        "FREQ:STOP?",
+        "FREQ:STOP %s",
+        "Control stop frequency in Hz.",
+    )
+
+    attenuation = Instrument.control(
+        "INP:ATT?",
+        "INP:ATT %s",
+        "Control attenuation in dB.",
+    )
+
+    res_bandwidth = Instrument.control(
+        "BAND:RES?",
+        # There is no space between RES and %s on purpose, see _number_or_auto.
+        "BAND:RES%s",
+        "Control resolution bandwidth in Hz. Can be set to 'AUTO'",
+        set_process=_number_or_auto,
+    )
+
+    video_bandwidth = Instrument.control(
+        "BAND:VID?",
+        "BAND:VID%s",
+        "Control video bandwidth in Hz. Can be set to 'AUTO'",
+        set_process=_number_or_auto,
+    )
+
+    # Sweeping -------------------------------------------------------------------------------------
+
+    sweep_time = Instrument.control(
+        "SWE:TIME?",
+        # No space between TIME and %s on purpose, see _number_or_auto.
+        "SWE:TIME%s",
+        "Control sweep time in s. Can be set to 'AUTO'.",
+        set_process=_number_or_auto,
+    )
+
+    continuous_sweep = Instrument.control(
+        "INIT:CONT?",
+        "INIT:CONT %s",
+        "Control continuous (True) or single sweep (False)",
+        validator=strict_discrete_set,
+        values={True: 1, False: 0},
+        map_values=True,
+    )
+
+    def single_sweep(self):
+        """Perform a single sweep with synchronization."""
+        self.write("INIT; *WAI")
+
+    def continue_single_sweep(self):
+        """Continue with single sweep with synchronization."""
+        self.write("INIT:CONM; *WAI")
+
+    # Traces ---------------------------------------------------------------------------------------
+
+    trace_mode = Instrument.control(
+        "DISP:TRAC:MODE?",
+        "DISP:TRAC:MODE %s",
+        "Control trace mode ('WRIT', 'MAXH', 'MINH', 'AVER' or 'VIEW')",
```

Comment:
You could add a description what each mode is.
Maybe with lower cases: `WRITe, AVERage`.
You can also use enums, if you wish.